lamindb_setup/_connect_instance.py

```diff
@@ -208,21 +219,29 @@ def connect(
                 return None
             else:
                 raise RuntimeError(MESSAGE_NO_MULTIPLE_INSTANCE)
-        elif settings._instance_exists and f"{owner}/{name}" != settings.instance.slug:
+        elif (
+            _write_settings
+            and settings._instance_exists
+            and f"{owner}/{name}" != settings.instance.slug
+        ):
             close_instance(mute=True)
 
         try:
-            isettings = _connect_instance(owner, name, db=db)
+            isettings = _connect_instance(owner, name, db=db, access_token=access_token)
         except InstanceNotFoundError as e:
             if _raise_not_found_error:
                 raise e
             else:
                 return "instance-not-found"
         if isinstance(isettings, str):
             return isettings
+        # after we checked that isettings is not a string
```

Comment:
The comment here is incomprehensible to me just because the wording is poor.

Reply:
edited